src/app/components/scan-fingers/scan-fingers.component.ts

```diff
@@ -1,20 +1,88 @@
-import { Component, OnInit } from '@angular/core';
-import { FingerprintReader } from '@digitalpersona/devices';
+import { Component, EventEmitter, Input, OnInit, Output } from '@angular/core';
+import {
+  FingerprintReader,
+  QualityCode,
+  QualityReported,
+  SampleFormat,
+} from '@digitalpersona/devices';
+
+enum ScanStatus{
+  Successful= 'Successful',
+  Failed = 'Failed'
+}
 
 @Component({
   selector: 'app-scan-fingers',
   templateUrl: './scan-fingers.component.html',
-  styleUrls: ['./scan-fingers.component.scss']
+  styleUrls: ['./scan-fingers.component.scss'],
 })
-export class ScanFingersComponent implements OnInit {
 
+export class ScanFingersComponent implements OnInit {
   private reader!: FingerprintReader;
 
-  constructor() { }
+  @Input() isReaderConnected: boolean;
+  @Output() isReaderConnectedChange = new EventEmitter<boolean>();
+
+  @Input() step: number;
+  @Output() stepChange = new EventEmitter<number>();
+
+  scanStatus?: ScanStatus;
```

Comment:
we need chain optional `?`?

Reply:
@hoangngk In new update, the status has value "Empty" if not yet scan.